lib/elixir/lib/enum.ex

```diff
@@ -2046,6 +2048,27 @@ defmodule Enum do
   end
 
   @doc """
+  Enumerates the collection, removing all duplicated items.
+
+  ## Examples
+
+      iex> Enum.uniq_by([{1, :x}, {2, :y}, {1, :z}], fn {x, _} -> x end)
+      [{1,:x}, {2,:y}]
+
+  """
+  @spec uniq_by(t, (element -> term)) :: list
+
+  # TODO: Use a HashSet
+  def uniq_by(collection, fun) when is_list(collection) do
+    do_uniq(collection, [], fun)
```

Comment:
This should use HashSet.new now (according to master). We can also remove the TODO.